traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Traveler.java

```diff
@@ -432,7 +432,8 @@ protected void main() {
     public static void fetchPurchaseForm(Product product, List<Pass> passes, FetchPurchaseFormCallback fetchPurchaseFormCallback) {
         if (!isInitialized()) return;
 
-        if (passes.size() == 0) {
+        boolean isPassesEmpty = (passes == null ||passes.size() == 0);
+        if (product.getProductType() == ProductType.BOOKABLE && isPassesEmpty) {
             fetchPurchaseFormCallback.onPurchaseFormFetchError(new PurchaseError(PurchaseError.Code.NO_PASSES));
             return;
```

Comment:
Should separate Parking and Bookable into separate methods

Reply:
✓